src/main/java/org/pkcs11/jacknji11/CKM.java

```diff
@@ -370,16 +373,29 @@
     }
 
     public long mechanism;
-    public byte[] pParameter;
+    public byte[] bParameter;
+    public Pointer pParameter;
+    public long ulParameterLen;
 
     /**
      * PKCS#11 CK_MECHANISM struct constructor.
      * @param mechanism CKM_? mechanism.  Use one of the public static final long fields in this class.
      * @param param param for mechanism
      */
-    public CKM(long mechanism, byte[] param) {
+    public CKM(long mechanism, Pointer param, int paramSize) {
```

Comment:
Is there any other code which calls this new constructor?

Reply:
We are using it for Utimaco CP5 operation. I plan to create pull requests for the Utimaco CP5 (eIDAS HSM) vendor mechanisms once these first batch of pull requests are accepted.

Example usage:
CKM mechanism = new CKM(CKM.CKM_CP5_INITIALIZE, params.getPointer(), params.size());

I plan to add java comments in the "CP5" pull request.